test/CSharpLiteWriterUnitTests/Given_an_OdcmClass_Entity_Collection_AddLinkAsync_Method.cs

```diff
@@ -0,0 +1,138 @@
+         * property called 'Friends'.
+         * 
+         * Spec - http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part1-protocol/odata-v4.0-errata02-os-part1-protocol-complete.html#_Toc406398333
+         */
+
+        [Fact]
+        public void It_adds_a_link_to_a_collection_valued_navigation_property()
+        {
+            var entityKeyValues = Class.GetSampleKeyArguments().ToArray();
+            var propertyPath = Class.GetDefaultEntityPath(entityKeyValues) + "/" + _navProperty.Name;
+            var entity2KeyValues = _navPropertyClass.GetSampleKeyArguments().ToArray();
+            var entity2Path = _navPropertyClass.GetDefaultEntityPath(entity2KeyValues);
+
+            using (_mockedService = new MockService())
+            {
+                var baseAddress = _mockedService.GetBaseAddress().TrimEnd('/');
+                var expectedJObject = new JObject();
+                expectedJObject["@odata.id"] = baseAddress + entity2Path;
+
+                _mockedService
+                    .SetupPostEntity(TargetEntity, entityKeyValues)
+                    .SetupPostEntity(_navPropertyEntity, entity2KeyValues)
+                    .OnPostAddLinkRequest(propertyPath, expectedJObject)
+                    .RespondWithODataOk();
+
+                var context = _mockedService.GetDefaultContext(Model);
+                var collection = context.CreateCollection(_navPropertyCollectionFetcherType, _navPropertyConcreteType,
+                    propertyPath);
+                var sourceInstance = context.CreateConcrete(ConcreteType);
+                var targetInstance = context.CreateConcrete(_navPropertyConcreteType);
+
+                collection.InvokeMethod<Task>("AddLinkAsync", new object[] { sourceInstance, targetInstance, System.Type.Missing }).Wait();
+            }
+        }
+
+        [Fact]
+        public void It_does_not_add_a_link_when_delay_saving()
+        {
+            var entityKeyValues = Class.GetSampleKeyArguments().ToArray();
+            var propertyPath = Class.GetDefaultEntityPath(entityKeyValues) + "/" + _navProperty.Name;
+            var entity2KeyValues = _navPropertyClass.GetSampleKeyArguments().ToArray();
+
+            using (_mockedService = new MockService()
+                    .SetupPostEntity(TargetEntity, entityKeyValues)
+                    .SetupPostEntity(_navPropertyEntity, entity2KeyValues))
+            {
+
+                var context = _mockedService.GetDefaultContext(Model);
+                var collection = context.CreateCollection(_navPropertyCollectionFetcherType, _navPropertyConcreteType,
+                    propertyPath);
+                var sourceInstance = context.CreateConcrete(ConcreteType);
+                var targetInstance = context.CreateConcrete(_navPropertyConcreteType);
+                
+                collection.InvokeMethod<Task>("AddLinkAsync", new object[] { sourceInstance, targetInstance, true }).Wait();
```

Comment:
We should test both sides of the delay save. This should be fairly easy:
- add a config setting to MockService that makes it throw if it gets an unexpected request
- Once you invoke the delaySave version of AddLinkAsync, setup the MockService to expect the AddLink request
- execute the batch